cmd/estuary-shuttle/rpc.go

```diff
@@ -315,58 +354,85 @@ func (s *Shuttle) trackTransfer(chanid *datatransfer.ChannelID, dealdbid uint) {
 	}
 }
 
-func (d *Shuttle) handleRpcStartTransfer(ctx context.Context, cmd *drpc.StartTransfer) error {
+func (s *Shuttle) handleRpcStartTransfer(ctx context.Context, cmd *drpc.StartTransfer) error {
+	ctx, span := s.Tracer.Start(ctx, "handleStartTransfer", trace.WithAttributes(
+		attribute.Int64("contentID", int64(cmd.ContentID)),
+		attribute.Int64("dealDbID", int64(cmd.DealDBID)),
+		attribute.String("miner", cmd.Miner.String()),
+		attribute.String("dataCID", cmd.DataCid.String()),
+		attribute.String("propCID", cmd.PropCid.String()),
+	))
+	defer span.End()
+
 	go func() {
-		chanid, err := d.Filc.StartDataTransfer(ctx, cmd.Miner, cmd.PropCid, cmd.DataCid)
+		chanid, err := s.Filc.StartDataTransfer(ctx, cmd.Miner, cmd.PropCid, cmd.DataCid)
 		if err != nil {
-			log.Errorf("failed to start requested data transfer: %s", err)
-			d.sendTransferStatusUpdate(ctx, &drpc.TransferStatus{
+			errMsg := fmt.Sprintf("failed to start data transfer: %s", err)
+			span.AddEvent(errMsg)
+			log.Error(errMsg)
+			s.sendTransferStatusUpdate(ctx, &drpc.TransferStatus{
 				DealDBID: cmd.DealDBID,
 				Failed:   true,
-				Message:  fmt.Sprintf("failed to start data transfer: %s", err),
+				Message:  errMsg,
 			})
 			return
 		}
 
-		d.trackTransfer(chanid, cmd.DealDBID)
+		s.trackTransfer(chanid, cmd.DealDBID)
 
-		if err := d.sendRpcMessage(ctx, &drpc.Message{
-			Op: drpc.OP_TransferStarted,
+		if err := s.sendRpcMessage(ctx, &drpc.Message{
+			TraceCarrier: drpc.NewTraceCarrier(span.SpanContext()),
```

Comment:
Here is one example of how the shuttle passes the trace context back to estuary.